packages/DApp/src/hooks/useCommunities.ts

```diff
@@ -23,57 +27,98 @@ export function useCommunities(publicKeys: string[]) {
-    publicKeys.forEach((publicKey, idx) => {
-      if (publicKey && communitiesHistories[idx]) {
-        const setCommunity = async () => {
-          const communityDetail = await getCommunityDetails(publicKey)
-          if (communityDetail) {
-            const communityHistory = communitiesHistories[idx]?.[0]
-            if (communityHistory && communityHistory.length > 0) {
-              const votingHistory = communityHistory.map((room: any) => {
-                const endAt = new Date(room.endAt.toNumber() * 1000)
-                return {
-                  ID: room.roomNumber.toNumber(),
-                  type: room.voteType === 1 ? 'Add' : 'Remove',
-                  result:
-                    endAt > new Date()
-                      ? 'Ongoing'
-                      : room.totalVotesFor.gt(room.totalVotesAgainst)
-                      ? 'Passed'
-                      : 'Failed',
-                  date: endAt,
-                }
-              })
-              dispatch({ ...communityDetail, votingHistory })
-            } else {
-              dispatch({ ...communityDetail, votingHistory: [] })
-            }
+    if (!waku || publicKeys.length === 0) return
+
+    const fetch = async () => {
+      await Promise.all(
+        publicKeys.map(async (publicKey) => {
+          const deserializedPublicKey = deserializePublicKey(publicKey)
+
+          if (communitiesDetails[deserializedPublicKey]) {
+            return
           }
-        }
-        setCommunity()
-      }
-    })
-  }, [JSON.stringify(publicKeys), communitiesHistories])
 
-  useEffect(() => {
-    setReturnCommunities(
-      publicKeys.map((publicKey) => {
-        const detail = communitiesDetails[publicKey]
-        if (detail) {
-          if (featureVotes[publicKey]) {
-            return { ...detail, featureVotes: featureVotes[publicKey].sum }
-          } else {
-            return { ...detail, featureVotes: BigNumber.from(0) }
+          const requestClient = new RequestClient(waku)
+          const community = await requestClient.fetchCommunityDescription(deserializedPublicKey)
+
+          if (!community) {
+            console.warn(`Community ${deserializedPublicKey} not found`)
+            return
           }
-        } else {
-          return undefined
-        }
-      })
-    )
-  }, [communitiesDetails, featureVotes, JSON.stringify(publicKeys)])
 
-  return returnCommunities
+          dispatch({
+            publicKey: deserializedPublicKey,
+            name: community!.identity!.displayName,
+            description: community!.identity!.description,
+            ens: community!.identity!.ensName,
+            icon: community!.identity!.images.large
+              ? URL.createObjectURL(
+                  new Blob([community!.identity!.images.large.payload], {
+                    type: 'image/jpeg',
+                  })
+                )
+              : null,
+            link: `https://join.status.im/c/${deserializedPublicKey}}`,
```

Comment:
non-blocking: seems like current site don't support this, see https://join.status.im/c/0x029f196bbfef4fa6a5eb81dd802133a63498325445ca1af1d154b1bb4542955133